tictactoe.rb

```diff
@@ -1,184 +1,185 @@
+				end
+				print_line += "\n"
 			end
-			print_line += "\n"
+			puts print_line
 		end
-		puts print_line
-	end
 
-	def make_move(who_player)
-    board_idx = condition_input
-    while !(valid_move?(board_idx[0], board_idx[-1]))
-			# if valid_move?(board_idx[0], board_idx[1])
-			# 	@board[board_idx[0]][board_idx[1]] = who_player
-			# else
-			board_idx = condition_input
-			valid_move?(board_idx[0], board_idx[-1])
+		def make_move(who_player)
+	    board_idx = condition_input
+	    while !(valid_move?(board_idx[0], board_idx[-1]))
+				# if valid_move?(board_idx[0], board_idx[1])
+				# 	@board[board_idx[0]][board_idx[1]] = who_player
+				# else
+				board_idx = condition_input
+				valid_move?(board_idx[0], board_idx[-1])
+			end
+			@board[board_idx[0]][board_idx[-1]] = who_player
 		end
-		@board[board_idx[0]][board_idx[-1]] = who_player
-	end
 
-	def condition_input
-		move = gets.chomp
-		move_a = move.split(' ')
-		row_word = move_a[0]
-		column_word = move_a[1]
-		return space(row_word, column_word)
-	end
-
-	def space(row, column)
-		case row
-		when "top"
-			first = 0
-		when "middle"
-			first = 1
-		when "bottom"
-			first = 2
-		else
-			first = 3
+		def condition_input
+			move = gets.chomp
+			move_a = move.split(' ')
+			row_word = move_a[0]
+			column_word = move_a[1]
+			return space(row_word, column_word)
 		end
 
-		case column
-		when "left"
-			second = 0
-		when "middle"
-			second = 1
-		when "right"
-			second = 2
-		else
-			second = 3
+		def space(row, column)
+			case row
+			when "top"
+				first = 0
+			when "middle"
+				first = 1
+			when "bottom"
+				first = 2
+			else
+				first = 3
+			end
+
+			case column
+			when "left"
+				second = 0
+			when "middle"
+				second = 1
+			when "right"
+				second = 2
+			else
+				second = 3
+			end
+				move_idx = [first, second]
+			return move_idx
 		end
-			move_idx = [first, second]
-		return move_idx
-	end
 
-	def valid_move?(row, column)
-
-	  if row == 3 || column == 3
-	  	puts "Please type: top middle bottom, with: left middle right"
-	  	return false
-	  elsif @board[row][column] != "-"
-			puts "You cannot overwrite another move. Try Again you fucking idiot."
-			return false
-	  else
-	  	return true
-	  end
-	end
+		def valid_move?(row, column)
+
+		  if row == 3 || column == 3
+		  	puts "Please type: top middle bottom, with: left middle right"
+		  	return false
+		  elsif @board[row][column] != "-"
+				puts "You cannot overwrite another move. Try Again you fucking idiot."
+				return false
+		  else
+		  	return true
+		  end
+		end
 
-	def winner
-		i = 0
-		winner_logic = false
-		while i < @board.length
-			if (@board[i][0] == @board[i][1]) && (@board[i][1] == @board[i][2]) && (@board[i][0] != "-")
-				# puts "#{@board[i][0]} is the winner!!!"
-				winner_logic = true
+		def winner
+			i = 0
+			winner_logic = false
+			while i < @board.length
+				if (@board[i][0] == @board[i][1]) && (@board[i][1] == @board[i][2]) && (@board[i][0] != "-")
```

Comment:
You may be able to use an enumerable here. Perhaps `rows = @board[i][0..-1]` and `rows.all?  = rows[i]` This is semi pseudo code and just possible food for though as an alternative way of acomplishing the same task which may be a bit more concise. I would commonly use `for,while,and until` statements but I think it is really much better if you can use `every,all,each` etc. Not that what you are doing is incorrect.

Reply:
Using `while,and for` statements would usually help me process the logic in my mind and step through things but you want to move from processing logic mentally to writing concise code.